drivers/sensor/icm42605/icm42605_setup.c

```
@@ -0,0 +1,377 @@
+
+	if (result) {
+		LOG_ERR("write REG_INTF_CONFIG1 failed");
+		return result;
+	}
+
+	v = BIT_EN_DREG_FIFO_D2A |
+	    BIT_TMST_TO_REGS_EN |
+	    BIT_TMST_EN;
+
+	result = inv_spi_single_write(REG_TMST_CONFIG, &v);
+
+	if (result) {
+		LOG_ERR("Write REG_TMST_CONFIG failed");
+		return result;
+	}
+
+	result = inv_spi_read(REG_INTF_CONFIG0, &v, 1);
+
+	if (result) {
+		LOG_ERR("Read REG_INTF_CONFIG0 failed");
+		return result;
+	}
+
+	LOG_DBG("Read REG_INTF_CONFIG0 0x%X", v);
+
+	/* 0x03 -> Disable i2c */
+	v |= 0x03;
+
+	result = inv_spi_single_write(REG_INTF_CONFIG0, &v);
+
+	if (result) {
+		LOG_ERR("Write REG_INTF_CONFIG failed");
+		return result;
+	}
+
+	v = 0;
+	result = inv_spi_single_write(REG_INT_CONFIG1, &v);
+
+	if (result) {
+		return result;
+	}
+
+	result = inv_spi_single_write(REG_PWR_MGMT0, &v);
+
+	if (result) {
+		return result;
+	}
+
+	return 0;
+}
+
+int icm42605_turn_on_fifo(struct device *dev)
+{
+	struct icm42605_data *drv_data = dev->data;
+
+	uint8_t int0_en, int1_en, fifo_en, smd;
+	uint8_t burst_read[3];
+	int r;
+	uint8_t v = 0;
+
+	v = BIT_FIFO_MODE_BYPASS;
+	r = inv_spi_single_write(REG_FIFO_CONFIG, &v);
+	if (r) {
+		return r;
+	}
+	v = 0;
+	r = inv_spi_single_write(REG_FIFO_CONFIG1, &v);
+	if (r) {
+		return r;
+	}
+	r = inv_spi_read(REG_FIFO_COUNTH, burst_read, 2);
+	if (r) {
+		return r;
+	}
+	r = inv_spi_read(REG_FIFO_DATA, burst_read, 3);
+	if (r) {
+		return r;
+	}
+	v = BIT_FIFO_MODE_STREAM;
+	r = inv_spi_single_write(REG_FIFO_CONFIG, &v);
+	if (r) {
+		return r;
+	}
+	int0_en = 0;
+	int1_en = 0;
+	smd = 0;
+
+	int0_en = BIT_INT_UI_DRDY_INT1_EN;
+	fifo_en |= (BIT_FIFO_ACCEL_EN  | BIT_FIFO_GYRO_EN | BIT_FIFO_WM_TH);
+
+	r = inv_spi_single_write(REG_FIFO_CONFIG1, &fifo_en);
+	if (r) {
+		return r;
+	}
+	r = inv_spi_single_write(REG_INT_SOURCE0, &int0_en);
+	if (r) {
+		return r;
+	}
+
+	if (drv_data->tap_en) {
+		v = BIT_TAP_ENABLE;
+		r = inv_spi_single_write(REG_APEX_CONFIG0, &v);
```

error is not checked here?

---

another comment: would be better to be consistent on either using `r` or `result`

---

Fixed